content/silver/concepts/automatic-attributes.md

```diff
@@ -159,11 +162,229 @@ top::Stmt ::= lifted::Decls
 ```
 
 A functor attribute is implemented as just an ordinary synthesized attribute whose type is the same as the type of the nonterminal on which it occurs.  To enable this, a functor attribute forwards to an ordinary synthesized attribute with a type parameter `a`.
-Functor attributes provide an overload for attribute occurrence such that `occurs on` for a functor attribute with no type argument provided will forward to will forward to an attribute occurrence with the nonterminal provided as the type argument.
+Functor attributes provide an overload for attribute occurrence such that `occurs on` for a functor attribute with no type argument provided will forward to an attribute occurrence with the nonterminal provided as the type argument.
```

Comment:
How about an example here. If a functor attribute specifies a type other than `a` when the attribute occurs on `a` should that be an error?

Reply:
I still don't understand this. Can one define a `functor` attribute with a type? What is an `occurs on` declaration without a type?

Reply:
I am talking about the type argument to the attribute in the occurs on declaration.  Did you see the paragraph and example I added below at the end of this section on functor attributes?